packages/rpc/src/WsClient.ts

```diff
@@ -100,44 +95,54 @@ export default class WsClient extends Client {
   }
 
   protected async send<const T extends RpcMethod>(
-    data: JsonRpcRequest<T>,
-  ): Promise<{ success: true; result: unknown } | { success: false; error: Error }> {
-    let requestId = data.id;
-
-    for (let i = 0; i < 5; i += 1) {
-      let socket;
-      try {
-        socket = await this.connectionReady();
-      } catch {
-        // retry
-        continue;
+    requests: JsonRpcRequest<T>[],
+  ): Promise<Response[]> {
+    const responses: Response[] = [];
+    for (const data of requests) {
+      const MAX_RETRIES = 5;
+      for (let i = 0; i < MAX_RETRIES; i += 1) {
+        let socket;
+        try {
+          socket = await this.connectionReady();
+        } catch {
+          // retry
+          continue;
+        }
```

Comment:
can probably move this outside the loop

Reply:
not really, we have a test to ensure we retry in case of connection failure

Reply:
yes but we probably don't need to attempt to reconnect 5 times per possible request. after we get the socket, sending the messages is synchronous so there is no time for the socket to move into an error state